### ISSUE TITLE
Make PG enums play nice with evolutions

### ISSUE THREAD

**Issue by strelec** [tminglei/slick-pg#74]
Play has this nice ability to automatically evolve the database with a click of a button:

http://www.playframework.com/documentation/2.0/resources/manual/detailledTopics/evolutions/images/evolutions.png

If you want to use ENUMs, you can't use this feature, since you get an error about the inexistent type on evolution.

Is it possible that enum evolutions would be generated along all others?

**Comment by vrcod** [comment]
@tminglei  Slick 3.0 is now released, which I understand contains major changes. Is it possible to somehow make code generator (with this plugin) to generate enum types automatically? For instance, JOOQ code generator can do this out of the box, so I'm guessing there's nothing preventing slick from doing the same either?

**Comment by tminglei** [comment]
Hi @vrcod, I'm not a fan of code generation.
I didn't find out how JOOQ to do it, can you help point it out?

**Comment by vrcod** [comment]
@tminglei From a perspective of someone who uses play evolutions, slick is only useful if it can generate the correct schema equivalent, that means including types such as Postgresql Enums, which is where your plugin comes in. JOOQ does generate them, an example looks like this: http://pastebin.com/xSxTyiqA
I can't find a link to a hard evidence stating an obvious fact, maybe this one will do: http://www.jooq.org/doc/2.6/manual/sql-execution/fetching/data-type-conversion/
From a practical perspective - editing slick table classes manually (without generator) equals to a nightmare when you have a database of some 500+ tables that is still in development.

**Comment by strelec** [comment]
It should be trivial, but make sure that you also include DROP TYPE statements before the CREATE TYPE ones, otherwise you get duplicates on every migration.

**Comment by vrcod** [comment]
@strelec if you're referring to play evolutions I'm more than familiar with it, my ups and downs are just file. Unless you mean something else here?

**Comment by strelec** [comment]
No, I am referring to a general problem of evolving enums, not addressing your post.
